src/v4/hooks/dca/DCAStructs.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {IAllowanceTransfer} from "permit2/src/interfaces/IAllowanceTransfer.sol";
+
+struct DCAIntent {
+    address swapper;
+    uint256 nonce;
+    uint256 chainId;
+    address hookAddress; // DCA contract's address
+    bool isExactIn; // EXACT_IN or EXACT_OUT
+    address inputToken; // Token to sell
+    address outputToken; // Token to buy
+    address cosigner; // TEE address that authorizes executions
+    uint256 minPeriod; // Minimum seconds between chunks
+    uint256 maxPeriod; // Maximum seconds between chunks
+    uint256 minChunkSize; // Min input or min output per chunk
+    uint256 maxChunkSize; // Max input or max output per chunk
+    uint256 minPrice; // Minimum price (output/input * 1e18)
+    uint256 deadline; // Intent expiration timestamp
+    OutputAllocation[] outputAllocations; // Distribution of output tokens
+    PrivateIntent privateIntent; // Private execution parameters - included in EIP-712 signature but zeroed on-chain, only hash revealed
+}
+
+struct PrivateIntent {
+    uint256 totalAmount; // Total amount on the exact side (input for EXACT_IN, output for EXACT_OUT)
+    uint256 exactFrequency;
+    uint256 numChunks;
+    bytes32 salt;
+    bytes32[] oracleFeeds; // Array of possible oracle feed IDs
+}
+
+struct OutputAllocation {
+    address recipient;
+    uint256 basisPoints; // Out of 10000 (100% = 10000)
+}
+
+struct DCAOrderCosignerData {
+    address swapper;
+    uint256 nonce;
```

Comment:
If you made `nonce` a uint96, we could pack it in the same slot as address (20 bytes).

Reply:
`execAmount` and `limitAmount` could be made `uint160` which is also the max size that permit2 transferFrom will accept.